server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -36,17 +38,23 @@ public class SourceFieldMapper extends MetadataFieldMapper {
     public static final String CONTENT_TYPE = "_source";
     private final XContentFieldFilter filter;
 
+    /** The source mode */
+    public enum Mode {
```

Comment:
does it need to be public considering that consumers rely mostly on the existing enabled() method?

Reply:
++, will make it private